### PR TITLE
205 rid of AFail/ASucc

### PR DESCRIPTION
**Pull request by prnvbn**
Replacing AFail/ASucc with [result](https://v2.ocaml.org/api/Result.html) which is part of the OCaml standard library.

Keeping with the spirit of this ticket, Success/Failure has been as well replaced. https://github.com/GillianPlatform/Gillian/blob/eaa028aaed7414c216b04ee25e0332dbc3d3cf8b/GillianCore/monadic/MonadicSMemory.ml#L20

### CODE REVIEW

**Review by giltho — CHANGES_REQUESTED** [review]
Thank you! Just a tiny change :)

File: Gillian-C/lib/MonadicSMemory.ml [hunk 1]
```diff
@@ -855,10 +855,7 @@ let get_print_info _ _ = (SS.empty, SS.empty)
 
 (* let str_noheap _ = "NO HEAP PRINTED" *)
 
-let lift_res res =
-  match res with
-  | Ok a -> Success a
-  | Error e -> Failure e
+let lift_res res = res
```
[the remark this answers is not in the document]
Comment (prnvbn) [reply Gillian-C/lib/MonadicSMemory.ml:858]: of course 🤦

**Comment by giltho** [comment]
Oh god completely forgot to merge this

**Comment by NatKarmios** [comment]
Resolves #205

**Comment by NatKarmios** [comment]
@prnvbn, pro tip, if you put "Resolves #xx" in your PR description, it'll automatically close the relevant issue when it's merged :)

See #156 for an example.

**Pull request merged**